packages/node_modules/@ciscospark/internal-plugin-avatar/src/avatar.js

```diff
@@ -36,8 +36,24 @@ const Avatar = SparkPlugin.extend({
    */
   _fetchAvatarUrl(uuid, options) {
     return this.store.get({uuid, size: options.size})
-      .catch(() => this.batcher.request({uuid, size: options.size}))
-        .then((item) => this.store.add(defaults({cacheControl: options.cacheControl}, item)));
+      .catch(() => Promise.all([
+        // fetch all possible sizes of avatar and store in cache
+        this.batcher.request({uuid, size: 40}),
```

Comment:
you'll need a set of unit tests that asserts the number of api calls that go out. Take a look at Xiaodan's PR that for how to spy on spark.request.